### PR TITLE
fix: add more logic on semver check of Starknet API

### PR DESCRIPTION
**Pull request by glihm**
# Description

## Tests

- [x] Yes
- [ ] No, because they aren't needed
- [ ] No, because I need help

## Added to documentation?

- [ ] README.md
- [ ] [Dojo Book](https://github.com/dojoengine/book)
- [x] No documentation needed

## Checklist

- [x] I've formatted my code (`scripts/prettier.sh`, `scripts/rust_fmt.sh`, `scripts/cairo_fmt.sh`)
- [x] I've linted my code (`scripts/clippy.sh`, `scripts/docs.sh`)
- [x] I've commented my code
- [x] I've requested a review after addressing the comments

### CODE REVIEW

**Review by lambda-0x — COMMENTED** [review]
wdyt about using: https://docs.rs/semver/latest/semver/struct.VersionReq.html#method.matches instead of doing that logic manually?

**Comment by kariy** [comment]
> wdyt about using: https://docs.rs/semver/latest/semver/struct.VersionReq.html#method.matches instead of doing that logic manually?

+1

**Comment by glihm** [comment]
> wdyt about using: https://docs.rs/semver/latest/semver/struct.VersionReq.html#method.matches instead of doing that logic manually?

Just modified with the use of matches, very good comment on that. Thanks for it.

File: bin/sozo/src/commands/migrate.rs [hunk 1]
```diff
@@ -167,3 +167,69 @@ pub async fn setup_env<'a>(
...
+
+    Ok(expected_ver_req.matches(&provided_ver))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_is_compatible_version_major_mismatch() {
```
Comment (lambda-0x) [inline bin/sozo/src/commands/migrate.rs:211]:
```suggestion
    fn test_is_incompatible_version_major_mismatch() {
```

typo?

**Review by lambda-0x — APPROVED** [review]
apart from above comment LGTM!

**Pull request merged**